android-sdk/src/test/java/org/sagebionetworks/bridge/android/data/ArchiveBuilderTest.java

```diff
@@ -0,0 +1,126 @@
+
+/**
+ * Created by jyliu on 1/30/2017.
+ */
+public class ArchiveBuilderTest {
+    private static final String APP_VERSION_NAME = "version 1.0, build 9";
+    private static final String DEVICE_NAME = "device";
+
+    private static final byte[] BYTES = new byte[]{1, 2, 3, 4};
+
+    @Mock
+    private BridgeConfig bridgeConfig;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        when(bridgeConfig.getDeviceName()).thenReturn(DEVICE_NAME);
+        when(bridgeConfig.getAppVersionName()).thenReturn(APP_VERSION_NAME);
+
+    }
+
+    @Test
+    public void testBuildSurvey() throws IOException {
+        String surveyGuid = "survey";
+        DateTime surveyCreatedOn = DateTime.now();
+
+
+        String filename1 = "file1";
+        DateTime endTime1 = DateTime.now().minusHours(1);
+        String json = "{'key' : 'value'}";
+
+        String filename2 = "file2";
+        DateTime endTime2 = DateTime.now().minusHours(2);
+        ByteSource byteSource = ByteSource.wrap(BYTES);
+
+        ArchiveFile file1 = new JsonArchiveFile(filename1, endTime1, json);
+        ArchiveFile file2 = new ByteSourceArchiveFile(filename2, endTime2, byteSource);
+
+        Archive archive = Archive.Builder.forSurvey(surveyGuid, surveyCreatedOn)
+                .withBridgeConfig(bridgeConfig)
+                .addDataFile(file1)
+                .addDataFile(file2)
+                .build();
+
+        ByteArrayOutputStream zipOutput = new ByteArrayOutputStream();
+        archive.writeTo(zipOutput);
+
+        Map<String, ByteArrayOutputStream> map = Maps.newHashMap();
+        byte[] buffer = new byte[1024];
+
+        ZipInputStream zis = new ZipInputStream(new ByteArrayInputStream(zipOutput.toByteArray()));
+        ZipEntry zipEntry = zis.getNextEntry();
+        try {
+            while (zipEntry != null) {
+                String fileName = zipEntry.getName();
+
+                ByteArrayOutputStream fileBytes = new ByteArrayOutputStream();
+                int len = 0;
+                while ((len = zis.read(buffer)) > 0) {
+                    fileBytes.write(buffer, 0, len);
+                }
+
+                map.put(fileName, fileBytes);
+                zis.closeEntry();
+                zipEntry = zis.getNextEntry();
+            }
+        } finally {
+            zis.close();
+        }
+
+        ByteArrayOutputStream bytes1 = map.get(filename1);
+        assertNotNull(bytes1);
+        assertEquals(json, new String(bytes1.toByteArray()));
+
+        ByteArrayOutputStream bytes2 = map.get(filename2);
+        assertNotNull(bytes2);
+        assertArrayEquals(BYTES, bytes2.toByteArray());
+
+        ByteArrayOutputStream bytesInfo = map.get("info.json");
+        assertNotNull(bytesInfo);
+
+        ArchiveInfo info = RestUtils.GSON.fromJson(bytesInfo.toString(), ArchiveInfo.class);
+
+        assertEquals(APP_VERSION_NAME, info.appVersion);
+        assertEquals(DEVICE_NAME, info.phoneInfo);
```

Comment:
Would it be worthwhile validating surveyGuid and surveyCreatedOn? What about a separate test validating item and schemaRevision?